### PR TITLE
Add Conversation API webhook MCP tools.

### PR DESCRIPTION
**Pull request by CarlosTardezAsin**
Expose list, get, create, update, and delete webhook tools via the Sinch SDK, with secrets stripped from responses and unit tests for formatting and handlers.

### CODE REVIEW

File: src/tools/conversation/utils/webhook-tools-helper.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Conversation } from '@sinch/conversation';
+
+export const appendRegionHint = (error: unknown, region: string): string => {
+  const message = error instanceof Error ? error.message : String(error);
+  return `${message} If the resource cannot be found, the region parameter may be incorrect. Current region: ${region}.`;
```
Comment (matsk-sinch) [inline src/tools/conversation/utils/webhook-tools-helper.ts]: nit: add a separator between the message and the hint. `${message}.`

File: src/tools/conversation/utils/webhook-tools-helper.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+export const dormantTriggersWarning =
```
Comment (matsk-sinch) [inline src/tools/conversation/utils/webhook-tools-helper.ts]: The warning is used both from `create-webhook` , and `update-webhook`, which the user configured something. Consider a neutral message:
"The webhook has no triggers and will remain..."

File: src/tools/conversation/utils/conversation-tools-helper.ts [hunk 3]
```diff
@@ -36,7 +36,27 @@ export const toolsConfig = defineToolsConfig({
...
+    tags: ['all', 'conversation', 'notification', 'create-webhook'],
+  },
+  updateWebhook: {
+    name: 'update-webhook',
+    tags: ['all', 'conversation', 'notification', 'update-webhook'],
+  },
+  deleteWebhook: {
+    name: 'delete-webhook',
+    tags: ['all', 'conversation', 'notification', 'delete-webhook'],
+  },
```
Comment (asein-sinch) [inline src/tools/conversation/utils/conversation-tools-helper.ts:75]: Can you replace `notification` by a new tag `configuration` ?

Here is the rational behind it:
 - all: all the tools 😎
 - conversation: all the tools related to the conversation domain (ie: conversation + templates + provisioning)
 - notification: all the tools for runtime notification - messaging, email, voice
 - configuration: all the tools to manage your Sinch products

The idea is to allow someone to limit the number of tools loaded in context depending on their use cases. We may change that in future version but let's roll with it for the moment

File: src/tools/conversation/utils/format-webhook-response.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+const formatWebhook = (webhook: Conversation.Webhook) => ({
+  id: webhook.id,
+  app_id: webhook.app_id,
+  target: webhook.target,
+  target_type: webhook.target_type,
+  triggers: webhook.triggers,
+});
+
+export const formatWebhookResponse = (webhook: Conversation.Webhook) =>
+  formatWebhook(webhook);
```
Comment (asein-sinch) [inline src/tools/conversation/utils/format-webhook-response.ts]: Is it really needed?

File: src/tools/conversation/utils/webhook-tools-helper.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  'No triggers were configured. The webhook will remain dormant until triggers are added (for example via the Sinch Dashboard).';
```
Comment (asein-sinch) [inline src/tools/conversation/utils/webhook-tools-helper.ts]: @matsk-sinch : Would it make sense to also add in the response that the tool 'update-webhook' can be used to configure the triggers on top of mentioning the dashboard only? And should we mention the dashboard at all, as the idea of the MCP server is to let an agent configure a Sinch account autonomously?

File: src/tools/conversation/update-webhook.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    return maybeService.promptResponse;
+  }
+  const conversationService = maybeService;
+  const usedRegion = setConversationRegion(region, conversationService);
+
+  const webhookUpdateRequestBody = {
+    ...(target !== undefined && { target }),
+    ...(triggers !== undefined && { triggers }),
+  } as Conversation.UpdateWebhookRequestBody;
+  const update_mask = buildUpdateMask(webhookUpdateRequestBody);
```
Comment (asein-sinch) [inline src/tools/conversation/update-webhook.ts]: The update mask is used to prevent the update of some parameters that are present in the request body. This is useful when the update request is formatted from the result of a GET operation. In the context of this function, you are forging the request from scratch from the request parameters, so the update_mask parameter is unnecessary.

So in the matter of keeping it simple, I'd suggest to remove it.

File: src/tools/conversation/utils/webhook-tools-helper.ts (hunk 1 above)
Comment (asein-sinch) [reply src/tools/conversation/utils/webhook-tools-helper.ts]: And maybe we can also return the list of the other available regions, to give to the LLM the opportunity to retry and learn to update the region in subsequent calls that involve this resource

File: tests/tools/conversation/list-webhooks.test.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  });
+
+  const result = await listWebhooksHandler({});
+
+  expect(mockList).toHaveBeenCalledWith({ app_id: 'app-123' });
+  expect(JSON.parse(result.content[0].text)).toEqual({
+    success: true,
+    webhooks: [{
+      id: 'wh-1',
+      app_id: undefined,
```
Comment (asein-sinch) [inline tests/tools/conversation/list-webhooks.test.ts]: Shouldn't it be `app-123` ?

File: src/tools/conversation/update-webhook.ts [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  target?: string;
+  triggers?: Conversation.WebhookTrigger[];
+  region?: string;
+}): Promise<IPromptResponse> => {
+  if (target === undefined && triggers === undefined) {
+    return new PromptResponse(JSON.stringify({
+      success: false,
+      error: 'At least one of target or triggers must be provided to update a webhook.',
+    })).promptResponse;
+  }
```
Comment (asein-sinch) [inline src/tools/conversation/update-webhook.ts:60]: We need to test the LLM behavior when we ask to remove the triggers, to make sure it will send an empty array as parameter.

File: src/tools/conversation/utils/webhook-tools-helper.ts (hunk 5 above)
Comment (matsk-sinch) [reply src/tools/conversation/utils/webhook-tools-helper.ts]: Good idea. We could even include `Use update-webhook with webhook_id=<id>...`. Mentioning the dashboard requires discussion. Since for signing the secret is out of scope because of elicitation. Let's discuss this offline.

File: src/tools/conversation/update-webhook.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+import { IPromptResponse, PromptResponse, Tags } from '../../types';
+import { isPromptResponse, matchesAnyTag } from '../../utils';
+
+const TOOL_KEY: ConversationToolKey = 'updateWebhook';
+const TOOL_NAME = getToolName(TOOL_KEY);
+
+export const registerUpdateWebhook = (server: McpServer, tags: Tags[]) => {
+  if (!matchesAnyTag(tags, toolsConfig[TOOL_KEY].tags)) return;
+
+  server.tool(
```
Comment (sergio-pekala) [inline src/tools/conversation/update-webhook.ts]: Lets change the .tool as it will be deprecated. Use .registerTool instead.

You can see in this pre-release how the already removed .tool, .resource and .prompt.
https://github.com/modelcontextprotocol/typescript-sdk/releases/tag/%40modelcontextprotocol%2Fserver%402.0.0-alpha.1

File: src/tools/conversation/resources/register-conversation-resources.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  'create-conversation-app',
+  'set-sms-channel-on-app',
+  'set-rcs-channel-on-app',
+  'set-whatsapp-channel-on-app',
+];
+
+export const registerConversationResources = (server: McpServer, tags: Tags[]) => {
+  if (!matchesAnyTag(tags, RESOURCE_TAGS)) return;
+
+  server.resource(
```
Comment (sergio-pekala) [inline src/tools/conversation/resources/register-conversation-resources.ts]: Lets change the .resource as it will be deprecated. Use .registerResource instead.

You can see in this pre-release how the already removed .tool, .resource and .prompt.
https://github.com/modelcontextprotocol/typescript-sdk/releases/tag/%40modelcontextprotocol%2Fserver%402.0.0-alpha.1

File: src/tools/conversation/create-conversation-app.ts [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+import { IPromptResponse, PromptResponse, Tags } from '../../types';
+import { ConversationRegionOverride } from './prompt-schemas';
+
+const TOOL_KEY: ConversationToolKey = 'createConversationApp';
+const TOOL_NAME = getToolName(TOOL_KEY);
+
+export const registerCreateConversationApp = (server: McpServer, tags: Tags[]) => {
+  if (!matchesAnyTag(tags, toolsConfig[TOOL_KEY].tags)) return;
+
+  server.tool(
```
Comment (sergio-pekala) [inline src/tools/conversation/create-conversation-app.ts]: same as commented for the other tool (apply to all the new tools please)

**Review by asein-sinch — APPROVED** [review]
